### PR TITLE
add tokenlon rfqv2 trades

### PR DESCRIPTION
**Pull request by 0xkeen**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by 0xkeen** [comment]
@Hosuke pls review

**Comment by Hosuke** [comment]
I am thinking if we can make the macro version of `tokenlon.trades_beta` in this PR.

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "user"                AS taker,
+        "maker"               AS maker,
+        "takerTokenAmount"    AS token_sold_amount_raw,
+        "makerTokenAmount"    AS token_bought_amount_raw,
+        cast(NULL as double)  AS amount_usd,
+        "takerToken"          AS token_sold_address,
+        "makerToken"          AS token_bought_address,
+        contract_address      AS project_contract_address,
+        evt_tx_hash           AS tx_hash,
+        CAST(ARRAY[] as array<bigint>) AS trace_address,
```
Comment (Hosuke) [inline models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql:34]: We may use CAST(ARRAY[-1] as array<bigint>) for null trace_address here:
```suggestion
        CAST(ARRAY[-1] as array<bigint>) as trace_address,
```

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+{% set blockchain = 'ethereum' %}
+{% set project = 'tokenlon' %}
+{% set project_version = '5' %}
+-- block
+{% set project_start_date = "timestamp '2023-09-12'" %}
+
+WITH dexs AS (
+    SELECT 
+        evt_block_time        AS block_time,
+        "user"                AS taker,
```
Comment (Hosuke) [inline models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql:25]: Is `"` needed for these fields? Or is it logically equivalent without `"`?
```suggestion
        user                AS taker,
```

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        "takerToken"          AS token_sold_address,
+        "makerToken"          AS token_bought_address,
+        contract_address      AS project_contract_address,
+        evt_tx_hash           AS tx_hash,
+        CAST(ARRAY[] as array<bigint>) AS trace_address,
+        evt_index
+    FROM
+        {{ source('tokenlon_v5_ethereum', 'RFQv2_evt_FilledRFQ') }}
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql:39]: We can use `incremental_predicate` here and in other incremental filters:
```suggestion
    WHERE {{ incremental_predicate('evt_block_time') }} 
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @0xkeen, do you want to add more seed records for tokenlon in this file:
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/aggregator/dex_aggregator_seed.csv

Then we can apply the `check_dex_aggregator_seed` test to `tokenlon_v5_ethereum_rfq_v2_trades`.

**Comment by 0xkeen** [comment]
@Hosuke the test case has passed, please review~

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @0xkeen 

[Sample tokenlon_v5_ethereum_rfq_v2_trades](https://dune.com/queries/3256317)

**Review by jeff-dude — APPROVED** [review]
thank you 🙏

**Pull request merged**